src/platforms/hosted/stlinkv2.c

```diff
@@ -676,20 +678,25 @@ static void stlink_mem_read(adiv5_access_port_s *ap, void *dest, uint32_t src, s
 {
 	if (len == 0)
 		return;
-	if (len > stlink.block_size) {
-		DEBUG_WARN("Too large!\n");
-		return;
-	}
 	if (!stlink_ensure_ap(ap->apsel))
 		raise_exception(EXCEPTION_ERROR, "ST-Link AP selection error");
 
 	uint8_t type;
-	if ((src & 1U) || (len & 1U))
+	uint16_t block_size;
+	if ((src & 1U) || (len & 1U)) {
 		type = STLINK_DEBUG_READMEM_8BIT;
-	else if ((src & 3U) || (len & 3U))
+		block_size = stlink.block_size;
+	} else if ((src & 3U) || (len & 3U)) {
 		type = STLINK_DEBUG_APIV2_READMEM_16BIT;
-	else
+		block_size = 4096;
+	} else {
 		type = STLINK_DEBUG_READMEM_32BIT;
+		block_size = 4096;
+	}
+	if (len > block_size) {
+		DEBUG_ERROR("%s(*ap, %p, 0x%08X, %lu): Too large! Must be <%u.\n", __func__, dest, src, len, block_size);
```

Comment:
Ah, this was written before -Wformat and before Windows builds were available and on CI. Agreed on `@0x2001ae80+768` format, `dest` on `hosted` can be anywhere, but I still would like to indicate AP number in this message (just in case). Updated. BMDA is not flash-constrained, and this line is not in hot/happy code path.
Can't easily test with `make CC=x86_64-w64-mingw32-gcc -C src/ PROBE_HOST=hosted HOSTED_BMP_ONLY=1 -j4` because it doesn't build with libusb or stlinkv2.o though. And the adapter doesn't support 64-bit [ADIv5, MEM-AP, ...] I think.